app/src/vanilla/java/cm/aptoide/pt/view/AppCoinsInfoPresenter.java

```diff
@@ -0,0 +1,89 @@
+package cm.aptoide.pt.view;
+
+import android.support.annotation.VisibleForTesting;
+import cm.aptoide.pt.app.view.AppCoinsInfoFragment;
+import cm.aptoide.pt.app.view.AppCoinsInfoView;
+import cm.aptoide.pt.crashreports.CrashReport;
+import cm.aptoide.pt.presenter.Presenter;
+import cm.aptoide.pt.presenter.View;
+import rx.Completable;
+import rx.Observable;
+
+/**
+ * Created by D01 on 02/08/2018.
+ */
+
+public class AppCoinsInfoPresenter implements Presenter {
+
+  private final AppCoinsInfoView view;
+  private final AppCoinsInfoNavigator appCoinsInfoNavigator;
+  private final AppCoinsInfoManager appCoinsInfoManager;
+  private final CrashReport crashReport;
+
+  public AppCoinsInfoPresenter(AppCoinsInfoView view, AppCoinsInfoNavigator appCoinsInfoNavigator,
+      AppCoinsInfoManager appCoinsInfoManager, CrashReport crashReport) {
+    this.view = view;
+    this.appCoinsInfoNavigator = appCoinsInfoNavigator;
+    this.appCoinsInfoManager = appCoinsInfoManager;
+    this.crashReport = crashReport;
+  }
+
+  @Override public void present() {
+    handleClickOnCoinbaseLink();
+    handleClickOnAppcWalletLink();
+    handleClickOnInstallButton();
+    handleButtonText();
+  }
+
+  @VisibleForTesting public void handleClickOnInstallButton() {
+    view.getLifecycle()
+        .filter(event -> event.equals(View.LifecycleEvent.CREATE))
+        .flatMap(__ -> Observable.merge(view.installButtonClick(), view.cardViewClick()))
+        .flatMap(click -> appCoinsInfoManager.loadButtonState())
+        .flatMapCompletable(isInstalled -> {
+          if (isInstalled) {
+            return openInstalledApp(AppCoinsInfoFragment.APPCWALLETPACKAGENAME);
```

Comment:
This makes the presenter depend on the interface implementation and not only the abstraction, should be injected in the presenter or passed in the click event.